packages/components/src/components/input-password/input-password-styles.ts

```diff
@@ -0,0 +1,167 @@
+  // ensuring that the paddingButton calculation solely depends on the scaling factor.
+  const paddingButton = compact ? '0px' : `calc(${formButtonOrIconPadding} * ${scalingVar})`;
+  const height = `max(${fontLineHeight}, ${scalingVar} * (${fontLineHeight} + 10px))`;
+
+  const { primaryColor, contrastLowColor, contrastMediumColor, disabledColor } = getThemedColors(theme);
+  const {
+    primaryColor: primaryColorDark,
+    contrastLowColor: contrastLowColorDark,
+    contrastMediumColor: contrastMediumColorDark,
+    disabledColor: disabledColorDark,
+  } = getThemedColors('dark');
+  const { formStateColor, formStateHoverColor } = getThemedFormStateColors(theme, state);
+  const { formStateColor: formStateColorDark, formStateHoverColor: formStateHoverColorDark } = getThemedFormStateColors(
+    'dark',
+    state
+  );
+
+  const hoverStyles = {
+    borderColor: formStateHoverColor || primaryColor,
+    ...prefersColorSchemeDarkMediaQuery(theme, {
+      borderColor: formStateHoverColorDark || primaryColorDark,
+    }),
+  };
+
+  return getCss({
+    '@global': {
+      ':host': {
+        display: 'block',
+        ...addImportantToEachRule({
+          ...colorSchemeStyles,
+          ...hostHiddenStyles,
+        }),
+      },
+      ...preventFoucOfNestedElementsStyles,
+      input: {
+        all: 'unset',
+        font: textSmallStyle.font.replace('ex', 'ex + 6px'), // a minimum line-height is needed for input, otherwise value is scrollable in Chrome, +6px is aligned with how Safari visualize date/time input highlighting
+        height,
+        paddingBlock,
+        textIndent: 0,
+        color: primaryColor,
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          color: primaryColorDark,
+        }),
+        minWidth: '2rem',
+        '&[type="text"]': {
+          textOverflow: 'ellipsis',
+        },
+      },
+    },
+    root: {
+      display: 'grid',
+      gap: spacingStaticXSmall,
+    },
+    wrapper: {
+      border: `${borderWidthBase} solid ${formStateColor || contrastMediumColor}`,
+      borderRadius: borderRadiusSmall,
+      paddingInlineStart: paddingInline,
+      paddingInlineEnd: paddingBlock, // `calc(${paddingInline} - (${paddingButton} * 2))`, TODO resolve inconsistency in Figma
+      display: 'grid',
+      gridTemplateColumns: '1fr auto',
+      gap: formElementLayeredGap,
+      transition: `${getTransition('background-color')}, ${getTransition('border-color')}, ${getTransition('color')}`,
+      ...prefersColorSchemeDarkMediaQuery(theme, {
+        borderColor: formStateColorDark || contrastMediumColorDark,
+      }),
+      '&:has(input:focus:not([readonly]))': {
+        borderColor: primaryColor,
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          borderColor: primaryColorDark,
+        }),
+      },
+      ...hoverMediaQuery({
+        ...(!disabled &&
+          !readOnly && {
+            '&:hover:not(:has(input:focus)):not(:has(p-button-pure:hover))': hoverStyles,
```

Comment:
You shouldn't use `p-button-pure` as selector since it won't work with prefixed components, use css class instead.